### PR TITLE
ResourceTypes, Primary and Misc Resources, Ingots, Dusts and Nuggets

### PR DESCRIPTION
**Pull request by TinyPandas**

### CODE REVIEW

File: src/main/java/net/panda/mekanismfabric/MekanismDataGenerator.java [hunk 1]
```diff
@@ -2,10 +2,224 @@
...
+
+//			CookingRecipeJsonBuilder.createSmelting(ingredient, RecipeCategory.MISC, output, 0.45F, 300)
+//					.criterion(FabricRecipeProvider.hasItem(),
+//							FabricRecipeProvider.conditionsFromItem(input))
+//					.offerTo(exporter, SMELTING_PATH + path + extractIdentifier(output));
+//
+//			CookingRecipeJsonBuilder.createSmelting(ingredient, RecipeCategory.MISC, output, 0.45F, 300)
+//					.criterion(FabricRecipeProvider.hasItem(input),
+//							FabricRecipeProvider.conditionsFromItem(input))
+//					.offerTo(exporter, BLASTING_PATH + path + extractIdentifier(output));
```
Comment (TinyPandas) [inline src/main/java/net/panda/mekanismfabric/MekanismDataGenerator.java]: Remove

File: src/main/java/net/panda/mekanismfabric/MekanismDataGenerator.java [hunk 2]
```diff
@@ -2,10 +2,224 @@
...
+			createCookingRecipe(exporter, List.of(input), output, path);
+		}
+
+		@Override
+		public void generate(Consumer<RecipeJsonProvider> exporter) {
+			// Block to Ingot
+			createShapelessResourceBlockToIngotRecipe(exporter, PrimaryResources.OSMIUM);
+			createShapelessResourceBlockToIngotRecipe(exporter, PrimaryResources.TIN);
+			createShapelessResourceBlockToIngotRecipe(exporter, PrimaryResources.LEAD);
+			createShapelessResourceBlockToIngotRecipe(exporter, PrimaryResources.URANIUM);
```
Comment (TinyPandas) [inline src/main/java/net/panda/mekanismfabric/MekanismDataGenerator.java]: Loop over `PrimaryResources` and call
- `createShapelessResourceBlockToIngotRecipe`
- `createShapelessIngotToNuggetRecipeFromResource`
- `createShapedIngotBlockRecipeFromResource`
- `createShapedNuggetToIngotRecipeFromResource`

File: src/main/java/net/panda/mekanismfabric/MekanismDataGenerator.java [hunk 3]
```diff
@@ -2,10 +2,224 @@
...
+
+		private void createShapedIngotToBlockRecipe(Consumer<RecipeJsonProvider> exporter, ItemConvertible input, ItemConvertible output) {
+			create3x3ShapedRecipe(exporter, input, output, RecipeCategory.BUILDING_BLOCKS, INGOT_TO_BLOCK_PATH);
+		}
+
+		private void createShapedNuggetToIngotRecipe(Consumer<RecipeJsonProvider> exporter, ItemConvertible input, ItemConvertible output) {
+			create3x3ShapedRecipe(exporter, input, output, RecipeCategory.MISC, NUGGET_TO_INGOT_PATH);
+		}
+
+		private void createShapelessResourceBlockToIngotRecipe(Consumer<RecipeJsonProvider> exporter, PrimaryResources resource) {
```
Comment (TinyPandas) [inline src/main/java/net/panda/mekanismfabric/MekanismDataGenerator.java]: Rename `createShapelessResourceBlockToIngotRecipe` to `createShapelessBlockToIngotRecipeFromResource`

File: src/main/java/net/panda/mekanismfabric/registries/MekanismBlocks.java [hunk 4]
```diff
@@ -1,25 +1,82 @@
...
+    public static final Block LEAD_ORE = registerBlock(new Identifier(MODID, "lead_ore"), new ExperienceDroppingBlock(FabricBlockSettings.copyOf(Blocks.IRON_ORE)));
+    public static final Block DEEPSLATE_LEAD_ORE = registerBlock(new Identifier(MODID, "deepslate_lead_ore"), new ExperienceDroppingBlock(FabricBlockSettings.copyOf(Blocks.IRON_ORE)));
+    public static final Block URANIUM_ORE = registerBlock(new Identifier(MODID, "uranium_ore"), new ExperienceDroppingBlock(FabricBlockSettings.copyOf(Blocks.IRON_ORE)));
+    public static final Block DEEPSLATE_URANIUM_ORE = registerBlock(new Identifier(MODID, "deepslate_uranium_ore"), new ExperienceDroppingBlock(FabricBlockSettings.copyOf(Blocks.IRON_ORE)));
+    public static final Block FLUORITE_ORE = registerBlock(new Identifier(MODID, "fluorite_ore"), new ExperienceDroppingBlock(FabricBlockSettings.copyOf(Blocks.IRON_ORE)));
+    public static final Block DEEPSLATE_FLUORITE_ORE = registerBlock(new Identifier(MODID, "deepslate_fluorite_ore"), new ExperienceDroppingBlock(FabricBlockSettings.copyOf(Blocks.IRON_ORE)));
```
Comment (TinyPandas) [inline src/main/java/net/panda/mekanismfabric/registries/MekanismBlocks.java]: Make universal `Identifier id(String name)` method.

File: src/main/java/net/panda/mekanismfabric/registries/MekanismWorldGen.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+import static net.panda.mekanismfabric.Mekanism.MODID;
+
+public class MekanismWorldGen {
+
+    private static final RegistryKey<PlacedFeature> FLUORITE_ORE_BURIED_PLACED_KEY = RegistryKey.of(RegistryKeys.PLACED_FEATURE, new Identifier(MODID, "fluorite_ore_buried"));
+
+    private static final RegistryKey<PlacedFeature> FLUORITE_ORE_NORMAL_PLACED_KEY = RegistryKey.of(RegistryKeys.PLACED_FEATURE, new Identifier(MODID, "fluorite_ore_normal"));
+
+    private static final RegistryKey<PlacedFeature> LEAD_ORE_NORMAL_PLACED_KEY = RegistryKey.of(RegistryKeys.PLACED_FEATURE, new Identifier(MODID, "lead_ore_normal"));
+
```
Comment (TinyPandas) [inline src/main/java/net/panda/mekanismfabric/registries/MekanismWorldGen.java]: Remove empty lines

File: src/main/java/net/panda/mekanismfabric/resource/ResourceTypes.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    DIRTY_DUST("dirty_dust"),
+    CLUMP("clump"),
+    INGOT("ingot"),
+    RAW("raw"),
+    NUGGET("nugget"),
+    ENRICHED("enriched");
+
+    private String pathPrefix;
+    private String pathSuffix;
+    private String path;
```
Comment (TinyPandas) [inline src/main/java/net/panda/mekanismfabric/resource/ResourceTypes.java]: No need to store `path` variable.

File: src/main/java/net/panda/mekanismfabric/resource/ResourceTypes.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    public String getPathPrefix() {
+        return pathPrefix;
+    }
+
+    public String getPathSuffix() {
+        return pathSuffix;
+    }
+
+    public String getPath() {
+        return path;
```
Comment (TinyPandas) [inline src/main/java/net/panda/mekanismfabric/resource/ResourceTypes.java]: Build `path` by `pathPrefix + "/" + pathSuffix`

File: src/main/resources/assets/mekanism-fabric/lang/en_us.json [hunk 8]
```diff
@@ -1,4 +1,109 @@
...
+  "block.mekanism-fabric.block_bronze": "Bronze Block",
+  "block.mekanism-fabric.block_refined_obsidian": "Refined Obsidian Block",
+  "block.mekanism-fabric.block_charcoal": "Charcoal Block",
+  "block.mekanism-fabric.block_refined_glowstone": "Refined Glowstone Block",
+  "block.mekanism-fabric.block_steel": "Steel Block",
+  "block.mekanism-fabric.block_fluorite": "Fluorite Block",
+  "block.mekanism-fabric.fluorite_ore": "Fluorite Ore",
+  "block.mekanism-fabric.deepslate_fluorite_ore": "Deepslate Fluorite Ore",
+  "block.mekanism-fabric.block_salt": "Salt Block",
+
```
Comment (TinyPandas) [inline src/main/resources/assets/mekanism-fabric/lang/en_us.json:27]: Remove additional empty line

File: src/main/resources/assets/mekanism-fabric/models/block/deepslate_ore/lead.json [hunk 9]
```diff
@@ -0,0 +1 @@
+{"parent":"minecraft:block/cube_all","textures":{"all":"mekanism-fabric:block/deepslate_lead_ore"}}
```
Comment (TinyPandas) [inline src/main/resources/assets/mekanism-fabric/models/block/deepslate_ore/lead.json]: Fix format

File: src/main/resources/assets/mekanism-fabric/models/block/deepslate_ore/osmium.json [hunk 10]
```diff
@@ -0,0 +1 @@
+{"parent":"minecraft:block/cube_all","textures":{"all":"mekanism-fabric:block/deepslate_osmium_ore"}}
```
Comment (TinyPandas) [inline src/main/resources/assets/mekanism-fabric/models/block/deepslate_ore/osmium.json]: Fix format

File: src/main/resources/assets/mekanism-fabric/models/block/deepslate_ore/tin.json [hunk 11]
```diff
@@ -0,0 +1 @@
+{"parent":"minecraft:block/cube_all","textures":{"all":"mekanism-fabric:block/deepslate_tin_ore"}}
```
Comment (TinyPandas) [inline src/main/resources/assets/mekanism-fabric/models/block/deepslate_ore/tin.json]: Fix format

File: src/main/resources/assets/mekanism-fabric/models/block/deepslate_ore/uranium.json [hunk 12]
```diff
@@ -0,0 +1 @@
+{"parent":"minecraft:block/cube_all","textures":{"all":"mekanism-fabric:block/deepslate_uranium_ore"}}
```
Comment (TinyPandas) [inline src/main/resources/assets/mekanism-fabric/models/block/deepslate_ore/uranium.json]: Fix format

File: src/main/resources/assets/mekanism-fabric/models/block/ore/lead.json [hunk 13]
```diff
@@ -0,0 +1 @@
+{"parent":"minecraft:block/cube_all","textures":{"all":"mekanism-fabric:block/lead_ore"}}
```
Comment (TinyPandas) [inline src/main/resources/assets/mekanism-fabric/models/block/ore/lead.json]: Fix format

File: src/main/resources/assets/mekanism-fabric/models/block/ore/osmium.json [hunk 14]
```diff
@@ -0,0 +1 @@
+{"parent":"minecraft:block/cube_all","textures":{"all":"mekanism-fabric:block/osmium_ore"}}
```
Comment (TinyPandas) [inline src/main/resources/assets/mekanism-fabric/models/block/ore/osmium.json]: Fix format

File: src/main/resources/assets/mekanism-fabric/models/block/ore/tin.json [hunk 15]
```diff
@@ -0,0 +1 @@
+{"parent":"minecraft:block/cube_all","textures":{"all":"mekanism-fabric:block/tin_ore"}}
```
Comment (TinyPandas) [inline src/main/resources/assets/mekanism-fabric/models/block/ore/tin.json]: Fix format

File: src/main/resources/assets/mekanism-fabric/models/block/ore/uranium.json [hunk 16]
```diff
@@ -0,0 +1 @@
+{"parent":"minecraft:block/cube_all","textures":{"all":"mekanism-fabric:block/uranium_ore"}}
```
Comment (TinyPandas) [inline src/main/resources/assets/mekanism-fabric/models/block/ore/uranium.json]: Fix format

File: src/main/resources/assets/mekanism-fabric/models/item/deepslate_lead_ore.json [hunk 17]
```diff
@@ -0,0 +1 @@
+{"parent":"mekanism-fabric:block/deepslate_ore/lead"}
```
Comment (TinyPandas) [inline src/main/resources/assets/mekanism-fabric/models/item/deepslate_lead_ore.json]: Fix format

File: src/main/resources/assets/mekanism-fabric/models/item/deepslate_osmium_ore.json [hunk 18]
```diff
@@ -0,0 +1 @@
+{"parent":"mekanism-fabric:block/deepslate_ore/osmium"}
```
Comment (TinyPandas) [inline src/main/resources/assets/mekanism-fabric/models/item/deepslate_osmium_ore.json]: Fix format

File: src/main/resources/assets/mekanism-fabric/models/item/deepslate_tin_ore.json [hunk 19]
```diff
@@ -0,0 +1 @@
+{"parent":"mekanism-fabric:block/deepslate_ore/tin"}
```
Comment (TinyPandas) [inline src/main/resources/assets/mekanism-fabric/models/item/deepslate_tin_ore.json]: Fix format

File: src/main/resources/assets/mekanism-fabric/models/item/deepslate_uranium_ore.json [hunk 20]
```diff
@@ -0,0 +1 @@
+{"parent":"mekanism-fabric:block/deepslate_ore/uranium"}
```
Comment (TinyPandas) [inline src/main/resources/assets/mekanism-fabric/models/item/deepslate_uranium_ore.json]: Fix format

File: src/main/resources/assets/mekanism-fabric/models/item/osmium_ore.json [hunk 21]
```diff
@@ -0,0 +1 @@
+{"parent":"mekanism-fabric:block/ore/osmium"}
```
Comment (TinyPandas) [inline src/main/resources/assets/mekanism-fabric/models/item/osmium_ore.json]: Fix format

File: src/main/resources/assets/mekanism-fabric/models/item/tin_ore.json [hunk 22]
```diff
@@ -0,0 +1 @@
+{"parent":"mekanism-fabric:block/ore/tin"}
```
Comment (TinyPandas) [inline src/main/resources/assets/mekanism-fabric/models/item/tin_ore.json]: Fix format

File: src/main/resources/assets/mekanism-fabric/models/item/uranium_ore.json [hunk 23]
```diff
@@ -0,0 +1 @@
+{"parent":"mekanism-fabric:block/ore/uranium"}
```
Comment (TinyPandas) [inline src/main/resources/assets/mekanism-fabric/models/item/uranium_ore.json]: Fix format

**Review by hero-ku — APPROVED** [review]

**Pull request merged**